src/AppBundle/Model/SlotCollectionDecorator.php

```diff
@@ -263,4 +340,47 @@ public function getContent()
         ksort($arr);
         return $arr;
     }
+
+    /**
+     * @inheritdoc
+     */
+    public function isLegalForMelee()
+    {
+        $slots = $this->getSlots()->getValues();
+        $cardCodes = [];
+        /**
+         * @var Decklistslot $slot;
+         */
+        foreach ($slots as $slot) {
+            $cardCodes[] = $slot->getCard()->getCode();
```

Comment:
Slight question for both this and the Joust variant - will this also pick up the agendas for a deck? It's not clear to me if agendas live in the "slots" along with the normal cards, or if they're separate.

I only ask because "The Wars To Come" agenda is on the Joust list.

Reply:
yes, that's accounted for. it picks up agendas as well. will put a test case deck up on sandbox...

Reply:
case in point: wamma's list https://sandbox.thronesdb.com/decklist/view/11/dr-goodstuff-s-green-bean-machine-12-0-euros-2018-1.0

Reply:
Awesome, thanks. 👍

Reply:
here it is: https://sandbox.thronesdb.com/decklist/view/12/illegal-deck-for-joust-1.0